query_utils.py

```diff
@@ -50,33 +57,34 @@ async def query_openai(
     for i in range(retry + 1):
         wait_time = (1 << min(i, OPENAI_EXP_CAP)) + random() / 10
         try:
+            logger.info("Attempting API call...")
             response = await openai.ChatCompletion.acreate(
                 model=model_name, messages=messages, **kwargs
             )
+            logger.info("API call successful")
+
+            # Log the request and response
             with open(HISTORY_FILE, "a") as f:
                 f.write(json.dumps((model_name, messages, kwargs, response)) + "\n")
+
             if any(choice["finish_reason"] != "stop" for choice in response["choices"]):
-                print("Truncated response!")
-                print(response)
+                logger.warning("Truncated response!")
+                logger.warning(response)
+
             contents = [choice["message"]["content"] for choice in response["choices"]]
             if n == 1:
                 return contents[0]
             else:
                 return contents
-        except (
-            openai.error.APIError,
-            openai.error.TryAgain,
-            openai.error.Timeout,
-            openai.error.APIConnectionError,
-            openai.error.ServiceUnavailableError,
-            openai.error.RateLimitError,
-        ) as e:
+        except openai.error.OpenAIError as e:
```

Comment:
I'm not particularly sure about this. For some errors like `AuthenticationError`, auto-retry doesn't make sense. What do you think?

Reply:
The reason I did that is that I was trying to make an API call with the wrong API key, and none of these errors detected the issues with the API that I tested because the code would be stuck in the TRY part, looping indefinitely in the following section:

```python
response = await openai.ChatCompletion.acreate(
                model=model_name, messages=messages, **kwargs
            )
``` 

In this case, I let OpenAI generate the error message by using the default error description I encountered while making an API call. At least, I could figure out my API problem by doing so while it was still running indefinitely in the response part.

Reply:
That's exactly my point. If it's an `AuthenticationError`, we shouldn't auto-retry, and let the error surface, right? Are you saying the original logic still catches wrong API key errors? If so, do you know why?

Reply:
The point is AuthenticationError - doesn not catch that error or any other errors related to API.